src/Aspire.Hosting.AWS/CDK/CDKLifecycleHook.cs

```diff
@@ -0,0 +1,66 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Aspire.Hosting.ApplicationModel;
+using Aspire.Hosting.AWS.CDK;
+using Aspire.Hosting.AWS.CloudFormation;
+using Aspire.Hosting.AWS.Utils;
+using Aspire.Hosting.Lifecycle;
+
+namespace Aspire.Hosting.AWS;
+
+/// <summary>
+/// The CDK Lifecycle Hook makes sure that the CDK resources are synthesized before publishing
+/// </summary>
+/// <param name="executionContext"></param>
+internal sealed class CDKLifecycleHook(DistributedApplicationExecutionContext executionContext) : IDistributedApplicationLifecycleHook
```

Comment:
Shouldn't this be done at some point in the `AWSProvisioner` lifecycle handler? Why do we need to have 2 separate lifecycle hooks?

Reply:
@normj The `AWSProvisioner` is used for provisioning the resources with local development where the CDKLifecycleHook makes sure the `StackArtifactResourceAnnotation` is set to the stack resources for both provisioning as for publishing (manifest creation). It more a separation of concerns thing. If it over complicates I'm happy to put it all in the `AWSProvisioner` and change that to `AWSLifecycleHook`

Reply:
I have merged the `CDKLifecycleHook` with the `AWSProvisioner` in the `AWSLifecycleHook`